### PR TITLE
Project dynamic router

### PR DESCRIPTION
**Pull request by 0xExp-po**
- add dynamic router
- fetch all project info with the project name from router param

### CODE REVIEW

**Comment by 0xExp-po** [comment]
- change the dynamic router to query params

File: dapp/src/pages/project/index.astro [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          typeof latestSha === "string" &&
+          latestSha.match(/^[a-f0-9]{40}$/)
+        ) {
+          setProjectLatestSha(latestSha);
+        } else setProjectLatestSha("");
+        initializeProjectState();
+        projectInfoLoaded.set(true);
+      } else {
+        refreshLocalStorage();
+        alert("There isn't such Project.");
```
Comment (tupui) [inline dapp/src/pages/project/index.astro:107]:
```suggestion
        alert("The project you are trying to access does not exist.");
```

**Review by tupui — COMMENTED** [review]
Thanks, this is great and works as expected.

It could go in if we want. The only thing I am not a fan of right now is the duplication we have with checking the `tomlData`. We have this logic in many other places. Can we have a function instead? Could be done now or later, let me know.

**Comment by 0xExp-po** [comment]
okay, I'll create a function and use it instead

**Pull request merged**